src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -71,6 +72,20 @@ void setConversationStore(ConversationStore conversationStore) {
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     List<Conversation> conversations = conversationStore.getAllConversations();
+
+    // Filter out test data to non-admins
+    HttpSession session = request.getSession();
+    boolean adminStatus = session.getAttribute("isAdmin") != null ? (boolean) session.getAttribute("isAdmin") : false;
+
+    if (adminStatus == false) {
+      TestDataGenerator tdg = TestDataGenerator.getInstance();
+      for (int i = conversations.size() - 1; i >= 0; i--) {
```

Comment:
Ah, interesting! Iterating backwards is a cute trick for allowing you to modify the list without messing up the indicies, but I think it would be clearer to read if you make a new list for the filtered data and copy the valid conversations into it. (It took me a minute to read through the code and understand why you couldn't just use a for-each loop here.)

Java8 streams are a nice, succinct way to do filtering like this--let me know if you'd like to talk through using them.!